src/nvim/edit.c

```diff
@@ -2658,6 +2658,31 @@ void ins_compl_show_pum(void)
   pum_selected_item = cur;
   pum_display(compl_match_array, compl_match_arraysize, cur, array_changed);
   curwin->w_cursor.col = col;
+
+  if (!has_event(EVENT_MENUPOPUPCHANGED)) {
+    return;
+  }
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+  if (cur < 0) {
+    tv_dict_add_dict(dict, S_LEN("completed_item"), tv_dict_alloc());
+  } else {
+    dict_T *item = ins_compl_dict_alloc(compl_curr_match);
+    tv_dict_add_dict(dict, S_LEN("completed_item"), item);
+  }
+  pumboundings_T *boundings = pum_get_boundings();
+  if (boundings != NULL) {
+    tv_dict_add_nr(dict, S_LEN("height"), boundings->height);
+    tv_dict_add_nr(dict, S_LEN("width"), boundings->width);
+    tv_dict_add_nr(dict, S_LEN("row"), boundings->row);
+    tv_dict_add_nr(dict, S_LEN("col"), boundings->col);
+    tv_dict_add_nr(dict, S_LEN("scrollbar"), boundings->scrollbar);
```

Comment:
`pum_get_boundings` could fill this part of the `dict`. Then we wouldn't need a new `pumboundings_T` type.

Reply:
The problem is I don't know how to extract data for the dict, `dict->width` doesn't work.